packages/kbn-server-route-repository/src/register_routes.test.ts

```diff
@@ -6,159 +6,169 @@
-      }),
-    });
-
-    registerRoutes({
-      core: coreSetup,
-      repository: {
-        'GET /internal/app/feature': {
-          endpoint: 'GET /internal/app/feature',
-          handler: internalHandler,
-          params: paramsRt,
-          options: internalOptions,
-        },
-        'GET /api/app/feature version': {
-          endpoint: 'GET /api/app/feature version',
-          handler: publicHandler,
-          params: paramsRt,
-          options: publicOptions,
-        },
-        'GET /internal/app/feature/error': {
-          endpoint: 'GET /internal/app/feature/error',
-          handler: errorHandler,
-          params: paramsRt,
-          options: internalOptions,
+  it('creates a router and defines the routes', () => {
+    callRegisterRoutes({
+      'POST /internal/route': {
+        endpoint: 'POST /internal/route',
+        handler: jest.fn(),
+        options: {
+          internal: true,
         },
       },
-      dependencies: {
-        aService: mockService,
+      'POST /api/public_route version': {
+        endpoint: 'POST /api/public_route version',
+        handler: jest.fn(),
+        options: {
+          public: true,
+        },
       },
-      logger: mockLogger,
     });
-  });
 
-  it('creates a router and defines the routes', () => {
     expect(createRouter).toHaveBeenCalledTimes(1);
 
-    expect(get).toHaveBeenCalledTimes(2);
+    expect(post).toHaveBeenCalledTimes(1);
 
-    const [internalRoute] = get.mock.calls[0];
-    expect(internalRoute.path).toEqual('/internal/app/feature');
-    expect(internalRoute.options).toEqual(internalOptions);
-    expect(internalRoute.validate).toEqual(routeValidationObject);
+    const [internalRoute] = post.mock.calls[0];
+    expect(internalRoute.path).toEqual('/internal/route');
+    expect(internalRoute.options).toEqual({
+      internal: true,
+    });
+    expect(internalRoute.validate).toEqual(noParamsValidationObject);
 
-    expect(getWithVersion).toHaveBeenCalledTimes(1);
-    const [publicRoute] = getWithVersion.mock.calls[0];
-    expect(publicRoute.path).toEqual('/api/app/feature');
-    expect(publicRoute.options).toEqual(publicOptions);
+    expect(postWithVersion).toHaveBeenCalledTimes(1);
+    const [publicRoute] = postWithVersion.mock.calls[0];
+    expect(publicRoute.path).toEqual('/api/public_route');
+    expect(publicRoute.options).toEqual({
+      public: true,
+    });
     expect(publicRoute.access).toEqual('public');
 
-    expect(getAddVersion).toHaveBeenCalledTimes(1);
-    const [versionedRoute] = getAddVersion.mock.calls[0];
+    expect(postAddVersion).toHaveBeenCalledTimes(1);
+    const [versionedRoute] = postAddVersion.mock.calls[0];
     expect(versionedRoute.version).toEqual('version');
     expect(versionedRoute.validate).toEqual({
-      request: routeValidationObject,
+      request: noParamsValidationObject,
     });
   });
 
+  it('does not allow any params if no schema is provided', () => {
```

Comment:
The test above checks that we're passing `noParamsValidationObject` to `validate` and this test just ensures that this schema is strict.